Sources/XcodeProj/Objects/Project/PBXProject.swift

```diff
@@ -548,7 +548,7 @@ extension PBXProject: PlistSerializable {
                 (value as? PBXObject)?.reference.value ?? value
             }
         }
-        plistAttributes[PBXProject.targetAttributesKey] = plistTargetAttributes
+        plistAttributes[PBXProject.targetAttributesKey] = plistTargetAttributes.isEmpty ? nil : plistTargetAttributes
```

Comment:
nit: might be useful to add a test to check the key isn't included when there are no attributes